lib/pure/bitops.nim

```diff
@@ -65,6 +65,9 @@ const useGCC_builtins = (defined(gcc) or defined(llvm_gcc) or
 const useICC_builtins = defined(icc) and useBuiltins
 const useVCC_builtins = defined(vcc) and useBuiltins
 const arch64 = sizeof(int) == 8
+const useBuiltinsRotate = (defined(amd64) or defined(i386)) and
+                          (defined(gcc) or defined(clang) or defined(vcc) or
+                           (defined(icl) and not defined(cpp))) and useBuiltins
```

Comment:
why `and not defined(cpp)`?

Reply:
also for intel compiler: is there any difference here between `icl` and `icc` defines?

Reply:
ICL does not have intrinsics when compiled in the C ++ backend (weird?).

The difference I know between ICL and ICC is that the first looks like Microsoft's CL (VCC) and the second looks like GCC. I downloaded the Intel compiler just for this PR. However, the installation of the intel compiler did not install ICC on my machine and I could not test it, so I did not add and also did not find any official documentation.